scio-test/src/main/scala/com/spotify/scio/testing/PipelineTestUtils.scala

```diff
@@ -173,4 +179,51 @@ trait PipelineTestUtils {
     val result: ScioResult = sc.run().waitUntilFinish() // block non-test runner
     (result, result.tap(f).value.toSeq)
   }
+
+  /**
+   * Generate an instance of `A` from `genA` using a random seed.
+   *
+   * @return
+   *   The result of `fn` applied to the generated `A`, or `None` on generation failure.
+   */
+  def withGen[A, T](genA: Gen[A])(
+    fn: A => T
+  )(implicit line: sourcecode.Line, name: sourcecode.Name): Option[T] =
+    withGen(genA, Seed.random())(fn)(line, name)
+
+  /**
+   * Generate an instance of `A` from `genA` using a the seed specified by `base64Seed`.
+   *
+   * @return
+   *   The result of `fn` applied to the generated `A`, or `None` on generation failure.
+   */
+  def withGen[A, T](genA: Gen[A], base64Seed: String)(
+    fn: A => T
+  )(implicit line: sourcecode.Line, name: sourcecode.Name): Option[T] =
+    withGen(genA, Seed.fromBase64(base64Seed).get)(fn)(line, name)
+
+  /**
+   * Generate an instance of `A` from `genA` using a the seed specified by `seed`.
+   *
+   * @return
+   *   The result of `fn` applied to the generated `A`, or `None` on generation failure.
+   */
+  def withGen[A, T](genA: Gen[A], seed: Seed)(
+    fn: A => T
+  )(implicit line: sourcecode.Line, name: sourcecode.Name): Option[T] = {
+    genA.apply(Gen.Parameters.default, seed) match {
+      case None =>
+        logger.error(
+          s"Failed to generate a valid value at ${name.value}:${line.value}. " +
+            s"Consider rewriting Gen instances to be less failure-prone. " +
+            s"Seed: ${seed.toBase64}"
+        )
+        None
+      case Some(a) =>
+        val r = Try(fn(a)).map(Some(_))
```

Comment:
should this be `.map(Option(_))` ? It's unlikely but a successful `fn` call could return null.

Reply:
In that case ... I think `null` would be the expected/correct return value?